c++/src/ColumnWriter.cc

```diff
@@ -1199,6 +1199,10 @@ namespace orc {
         tsStats->increase(1);
         tsStats->update(millsUTC);
 
+        if (secs[i] < 0 && nanos[i] != 0) {
+          secs[i] += 1;
```

Comment:
If secs[i] is -1 and we add 1 to it here. On the reader side, the if branch will simply fail to minus 1:
` if (secsBuffer[i] < 0 && nanoBuffer[i] != 0) { secsBuffer[i] -= 1; } `  This results in an 1 second shift of the original data. I canot find any good workaround but throw an exception if -1 has been found. @rip-nsk @omalley

Reply:
possible solutions 
- allow negative nanos in orc stream to cover the second before unix epoch.
- introduce new Timestamp type, w/o shift and deprecate old one